### ISSUE TITLE
ROG Flow Z13 When powered on, press any key on the keyboard to turn on airplane mode. The keyboard backlight turns off automatically after a certain period of time

### ISSUE THREAD

**Issue by Ernest-L** [seerge/g-helper#1660]
When ROG Flow Z13 is powered on, press any button to turn on airplane mode.
The keyboard backlight will automatically turn off after a period of time, and the keyboard backlight can only be turned on again through shortcut keys.

Turning on the ASUS Optimization service can solve the above problem, but if you manually drag the backlight brightness in G-Helper, it will cause the backlight to not match the prompts. For example, the window prompts the brightness to be maximum, but the actual keyboard backlight goes out.
[log.txt](https://github.com/seerge/g-helper/files/13469623/log.txt)

**Comment by Ernest-L** [comment]
![20231127140206](https://github.com/seerge/g-helper/assets/59217607/85fa769b-1220-4321-9270-9da0ee126e7f)
After turning on ASUS Optimization, the backlight shortcut keys are abnormal, as shown in the figure. The keyboard backlight goes out and prompts Low. The log of pressing the shortcut keys is as follows:


2023/11/27 14:08:59: Key: 199
2023/11/27 14:08:59: WMI event 199

**Comment by seerge** [comment]
@Ernest-L "Airplane mode" Fn+F12 is not handled by the app at all, it's it emits windows shortcut to turn airplane mode in windows itself on a hardare level

As for backlight brightness control, i see that you press a button and app sends corresponding command to your keyboard, so it's all working
```
2023/11/27 11:35:44: Key: 199
2023/11/27 11:35:44: HotKey 18C6: 5D-BA-C5-C4-03
2023/11/27 11:35:44: HotKey 1A30: 5D-BA-C5-C4-03
```

If you want to disable backlight timeout, you can do that under Extra settings by setting it to 0

**Comment by Ernest-L** [comment]
thank you for your reply. When I turn on "ASUS Optimization Service", the airplane mode issue is resolved, but it seems that "G-Helper" conflicts with "ASUS Optimization Service". When I hold down the button (FN+F11 to adjust backlight brightness), it prompts for the backlight brightness level, but when I release the button, it prompts again for the next backlight brightness level. This may cause the prompt to differ from the actual brightness. Prompt backlight brightness "low". In fact, the keyboard backlight is turned off. Can the backlight brightness prompt be turned off?

**Comment by seerge** [comment]
I would not advice to run optimization service along the app, as it simply conflicts and tries to do same things as app is doing :)

**Comment by seerge** [comment]
If those things happen with optimization service OFF, it means you have other programs doing that, including turning your backlight off. 

I would recommend to check what you have installed and running.

Anyway, it's not related to g-helper anyhow.

**Comment by Ernest-L** [comment]
As you said, the problem has nothing to do with G-Helper. During this period, I reinstalled a clean Windows 11 without ASUS software or G-Helper. The above problem still exists. But I seem to have discovered a pattern:
Regarding question 1, I tried to turn off "ASUS Optimization Service" and G-Helper's self-starting. After restarting the computer, this problem will persist until I manually turn on G-Helper, and the keyboard is normal. Therefore, I believe that ASUS's "ASUS Optimization Service" is a system service. It starts automatically when the system starts, so there will be no problems. G-Helper uses the task plan to start automatically after logging in, so an exception will occur on the lock screen interface. After unlocking and logging in to Windows, G-Helper completes the startup and the keyboard returns to normal.

Regarding question 2, ROG Flow Z13 is a 2-in-1 notebook similar to Surface Pro. After it is turned on, even if the backlight is set to always on, the backlight will go out after a period of time, and the backlight can only be turned on by readjusting the brightness. However, when the keyboard is unplugged and reconnected after turning on the computer, the backlight is normal and will not turn off automatically.

The keyboard firmware has been upgraded to the latest firmware ROGKBFWUpdateTool_OneKey_3.7.0.001. The firmware description indicates that the backlight problem has been solved, but the backlight problem still exists after the upgrade.

For these two problems, they can be solved after enabling ASUS's "ASUS Optimization Service", but they will conflict with G-Helper. I personally feel that ASUS officially solved the above two problems through keyboard firmware and optimization services, but G-Helper did not add this part of optimization, resulting in the above two problems.

This is what I got through experiments. I look forward to your reply and help.
